internal/transformer/classes/generators/dockerfilegenerators/utils.go

```diff
@@ -17,15 +17,108 @@
 package dockerfilegenerators
 
 import (
+	"bufio"
+	"fmt"
+	"os"
 	"path/filepath"
+	"regexp"
+	"strings"
 
 	"github.com/konveyor/move2kube/environment"
 	"github.com/konveyor/move2kube/internal/common"
+	"github.com/konveyor/move2kube/types/source/dotnet"
 	transformertypes "github.com/konveyor/move2kube/types/transformer"
 	"github.com/konveyor/move2kube/types/transformer/artifacts"
 	"github.com/sirupsen/logrus"
 )
 
+// isSilverlight tests whether the app is a sliverlight web application
+func isSilverlight(configuration dotnet.CSProj) (bool, error) {
+	if configuration.ItemGroups == nil || len(configuration.ItemGroups) == 0 {
+		return false, fmt.Errorf("No item groups in project file to parse")
+	}
+
+	for _, ig := range configuration.ItemGroups {
+		if ig.Contents == nil || len(ig.Contents) == 0 {
+			continue
+		}
+
+		for _, r := range ig.Contents {
+			if dotnet.WebSLLib.MatchString(r.Include) {
+				return true, nil
+			}
+		}
+	}
+
+	return false, nil
+}
+
+// isWeb tests whether the project is a web application
+func isWeb(configuration dotnet.CSProj) (bool, error) {
+	if configuration.ItemGroups == nil || len(configuration.ItemGroups) == 0 {
+		return false, fmt.Errorf("No item groups in project file to parse")
+	}
+
+	for _, ig := range configuration.ItemGroups {
+		if ig.References == nil || len(ig.References) == 0 {
+			continue
+		}
+
+		for _, r := range ig.References {
+			if dotnet.WebLib.MatchString(r.Include) {
+				return true, nil
+			}
+		}
+	}
+
+	return false, nil
+}
+
+// parseSolutionFile parses the solution file for the project file path
+func parseSolutionFile(inputPath string) []string {
+	solFile, err := os.Open(inputPath)
+	if err != nil {
+		logrus.Errorf("Could not open the solution file: %s", err)
+		return nil
+	}
+	defer solFile.Close()
+
+	solFileScanner := bufio.NewScanner(solFile)
+
+	r := regexp.MustCompile(dotnet.ProjBlock)
+
+	csr := regexp.MustCompile(dotnet.CsProj)
+
+	projectPaths := make([]string, 0)
+	for solFileScanner.Scan() {
+		s := solFileScanner.Text()
+		if r.MatchString(s) {
+			tokens := strings.Split(s, "=")
+			if len(tokens[1]) > 0 {
+				values := strings.Split(tokens[1], ",")
+				for _, v := range values {
+					if csr.MatchString(v) {
+						projectPaths = append(projectPaths, v)
+					}
+				}
+			}
+		}
+	}
+
+	if err := solFileScanner.Err(); err != nil {
+		logrus.Errorf("Could not parse the solution file: %s", err)
+		return nil
+	}
+
```

Comment:
Won't this depend on the is the progem is in? If so, we should avoid hardcoding the slash type

Reply:
@ashokponkumar the relative path that comes from the csproj file has windows-slashes...but if i am running move2kube on mac, but trying to read the csproj, then I get the base path in mac-slash format, and the relative path in win-slash format...hence converting them both to a single format to ease-of-use. Is this approach ok?